api/src/main/java/org/openmrs/util/HttpUrl.java

```diff
@@ -25,8 +25,8 @@
 	public HttpUrl(String url) throws MalformedURLException {
 		if (url == null) {
 			throw new MalformedURLException("Url cannot be null");
-		} else if (!url.startsWith("http://")) {
-			throw new MalformedURLException("Not a valid http url");
+		} else if (!url.startsWith("http://") && !url.startsWith("https://")) {
+			throw new MalformedURLException("Not a valid http(s) url");
```

Comment:
Will this ever accept any of http:// or https://?

Reply:
It will catch other urls like ftp:// or more likely throw when people forget to include http and just do www.openmrs.... etc.

There should be tests that exercise it.